torch/csrc/api/include/torch/detail/ordered_dict.h

```diff
@@ -0,0 +1,204 @@
+#pragma once
+
+#include <ATen/Error.h>
+
+#include <cstdint>
+#include <functional>
+#include <initializer_list>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+namespace torch {
+namespace detail {
```

Comment:
Any particular reason this is in detail?

Reply:
I didn't really foresee users of the C++ API needing to use it, since it's more of an implementation detail